src/views/Register.vue

```diff
@@ -49,18 +51,31 @@ export default class Register extends Vue {
     confirmPassword = '';
     email = ''
 
-    checkCompromisedPassword(): boolean {
+    async checkCompromisedPassword(): Promise<boolean | undefined> {
         const hashedPassword: string = sha1(this.password);
-        // todo send api 
-        return true
+        try {
+            const response = await axios.post(`${GarnBarnApiConfig.apiPrefix}/api/v1/account/compromised`, { hashedPassword });
+            if (response.status == 200) {
+                return true;
+            }
+        } catch (error: any) {
+            if (error.response.status == 302) {
+                return false;
+            }
+            return undefined
+        }
+        return false
     }
 
-    register(): void {
-        if (!this.checkCompromisedPassword()) {
+    async register(): Promise<void> {
+        const checkingPasssowrd = await this.checkCompromisedPassword()
+        if (!checkingPasssowrd) {
```

Comment:
Why the logic is not straight  forward? `True` = compromised, `False` = NOT compromised